src/Orchard.DisplayManagement/Razor/ThemeAwareViewLocationExpander.cs

```diff
@@ -16,50 +16,67 @@ public class ThemeAwareViewLocationExpander : IViewLocationExpander
         public virtual IEnumerable<string> ExpandViewLocations(ViewLocationExpanderContext context,
                                                                IEnumerable<string> viewLocations)
         {
-            var extensionManager = context.ActionContext.HttpContext.RequestServices.GetService<IExtensionManager>();
-            var themeManager = context.ActionContext.HttpContext.RequestServices.GetService<IThemeManager>();
-            var siteService = context.ActionContext.HttpContext.RequestServices.GetService<ISiteService>();
-            var site = siteService.GetSiteSettingsAsync().GetAwaiter().GetResult();
+            var result = new List<string>();
 
-            var orderedFeatures = extensionManager.GetFeatures();
-            var orderedExtensions = orderedFeatures.Where(f => f.Id == f.Extension.Id).Select(f => f.Extension);
-            var extension = orderedExtensions.FirstOrDefault(e => e.Id == context.AreaName);
+            var themeManager = context
+                .ActionContext
+                .HttpContext
+                .RequestServices
+                .GetService<IThemeManager>();
 
-            var themes = new List<IExtensionInfo>();
-            var currentTheme = themeManager.GetThemeAsync().GetAwaiter().GetResult();
-            var adminThemeId = (string)site.Properties["CurrentAdminThemeName"];
+            if (themeManager != null)
+            {
+                var extensionManager = context
+                    .ActionContext
+                    .HttpContext
+                    .RequestServices
+                    .GetService<IExtensionManager>();
 
-            themes.Add(currentTheme);
-            themes.AddRange(GetBaseThemes(currentTheme, adminThemeId, orderedExtensions));
-            var result = new List<string>();
+                var siteService = context
+                    .ActionContext
+                    .HttpContext
+                    .RequestServices
+                    .GetService<ISiteService>();
 
-            foreach (var theme in themes)
-            {
-                var themeViewsPath = Path.Combine(Path.DirectorySeparatorChar + theme.SubPath,
-                    "Views", context.AreaName != theme.Id ? context.AreaName : String.Empty);
+                var themes = new List<IExtensionInfo>();
 
-                result.Add(Path.Combine(themeViewsPath, "{1}", "{0}.cshtml"));
-                result.Add(Path.Combine(themeViewsPath, "Shared", "{0}.cshtml"));
-            }
+                var currentTheme = themeManager.GetThemeAsync().GetAwaiter().GetResult();
+                themes.Add(currentTheme);
 
-            if (extension != null)
-            {
-                var extensionViewsPath = Path.Combine(Path.DirectorySeparatorChar
-                    + extension.SubPath, "Views");
+                var site = siteService.GetSiteSettingsAsync().GetAwaiter().GetResult();
+                var adminThemeId = (string)site.Properties["CurrentAdminThemeName"];
```

Comment:
If we are in the Admin, then the admin theme is the value of the previous `currentTheme`, I don't think we need to special-case the admin theme.